src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -179,6 +181,14 @@ internal static int GetTenBaseNumberFromOctalAsciiChars(Span<byte> buffer)
             return string.IsNullOrEmpty(str) ? 0 : Convert.ToInt32(str, fromBase: 8);
         }
 
+        // Receives a byte array that represents an ASCII string containing a number in octal base.
+        // Converts the array to an octal base number, then transforms it to ten base and returns it.
+        internal static long GetTenBaseLongFromOctalAsciiChars(Span<byte> buffer)
+        {
+            string str = GetTrimmedAsciiString(buffer);
```

Comment:
Do we really need to create an intermediate string just to parse it into an integer?

Reply:
I don't see a Utf8Parser API that parses the integer from an "octal" string. I see it supports `x` - hexstring.

Reply:
I agree that a new allocation is not needed. This code can be improved to instead of returning a string, it returns an int representing the length of the ROS for slicing.

The method checks that the last character(s) in the ROS are either a `0` (null char) or a `32` (space). All other characters are not trimmed. Which means that if an unexpected non-numeric character is found, it will cause the conversion to fail.

Do you mind if I address this request later? I'd like to get this PR merged just for the DateTimeOffsets.

Reply:
> Do you mind if I address this request later?

I think that is fine.